src/portable/portable.c

```diff
@@ -384,6 +390,21 @@ static int portable_extract_by_path(
                 _cleanup_(rmdir_and_freep) char *tmpdir = NULL;
                 _cleanup_close_pair_ int seq[2] = EBADF_PAIR;
                 _cleanup_(sigkill_waitp) pid_t child = 0;
+                DissectImageFlags flags =
+                        DISSECT_IMAGE_READ_ONLY |
+                        DISSECT_IMAGE_GENERIC_ROOT |
+                        DISSECT_IMAGE_REQUIRE_ROOT |
+                        DISSECT_IMAGE_DISCARD_ON_LOOP |
+                        DISSECT_IMAGE_RELAX_VAR_CHECK |
+                        DISSECT_IMAGE_USR_NO_ROOT |
+                        DISSECT_IMAGE_ADD_PARTITION_DEVICES |
+                        DISSECT_IMAGE_PIN_PARTITION_DEVICES |
+                        DISSECT_IMAGE_ALLOW_USERSPACE_VERITY;
+
+                if (path_is_extension)
```

Comment:
these used to be passed only to dissected_image_mount(), now they are also passed to dissect_loop_device(), is that ok to do?

Reply:
and viceversa - lots more flags passed to dissected_image_mount() now

Reply:
yes, this is safe, and we generally should do this. Some of the flags are interpreted by one, and some by the other, and some by both (the latter is the rdonly flag for example). Generally if you pass it to one, you should also pass it to the other.

Look for example the all-in-one helper we use for things like tmpfiles.c, sysusers.c and so on: mount_image_privately_interactively(). It passes the same flags to a bunch of functions of dissect-images.c, including those we pass here.